examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts

```diff
@@ -2,32 +2,71 @@ import { Injectable } from '@nestjs/common';
 import { Witness } from '../../../models/witness';
 import { Proof } from '../../../models/proof';
 import { ICircuitService } from '../circuit.interface';
+import * as snarkjs from 'snarkjs';
+import * as verificationKey from '../../../../../../zeroKnowledgeKeys/circuit/circuit_verification_key.json';
+import 'dotenv/config';
 
 @Injectable()
 export class SnarkjsCircuitService implements ICircuitService {
-  public async createWitness(input: object): Promise<Witness> {
-    const witness: Witness = {
-      proof: { a: ['a'], b: [['b']], c: ['c'] },
-      publicInput: ['publicInput'],
-      verificationKey: 'verificationKey',
-    };
-
-    return witness;
+  public witness: Witness;
+
+  public async createWitness(inputs: object): Promise<Witness> {
+    this.witness = new Witness();
+
+    const proof = await this.createProof(inputs);
+    this.witness.proof = proof;
+
+    const publicInputs = await this.getPublicInputs(inputs);
```

Comment:
@biscuitdey `getPublicInputs` runs `executeCircuit`, but why since `createProof` already executes the circuit. Should it not be simply:
```
const publicInputs = this.witness.proof.a
```
?

Reply:
The snarkjs lib function for executing circuit returns the proof and public inputs together. I have added different functions to separate out these 2 values. Also, `this.witness.proof.a` is an element on the G1 elliptic curve. Public inputs values are different from this.

Reply:
yeah sorry a bit blind @biscuitdey 

can we not still do the following:
```
const { proof, publicInputs } = await this.executeCircuit(inputs);

const newProof = {
      a: proof.pi_a,
      b: proof.pi_b,
      c: proof.pi_c,
      protocol: proof.protocol,
      curve: proof.curve,
    };

this.witness.proof = newProof;
```

and remove `createProof` and `getPublicInputs` functions?

Reply:
sure I'll do that.